code/math/vecmat.cpp

```diff
@@ -808,44 +808,21 @@ matrix *vm_vector_2_matrix_norm(matrix *m, const vec3d *fvec, const vec3d *uvec,
 }
 
 
-//rotates a vector through a matrix. returns ptr to dest vector
-//dest CANNOT equal source
-//
-// Goober5000: FYI, the result of rotating a normalized vector through a rotation matrix will
-// also be a normalized vector.  It took me awhile to verify online that this was true. ;)
+// rotates a vector through a matrix, writes to *dest and returns the pointer
 vec3d *vm_vec_rotate(vec3d *dest, const vec3d *src, const matrix *m)
 {
-	Assert(dest != src);
```

Comment:
I'm not sure I follow, the assert was removed because the operation no longer requires it to be the case, its totally safe to have them be the same.

Reply:
Ok, I see the copies.  Did we profile these to make sure we don't have a performance hit by having to copy them?

Reply:
I haven't profiled it beyond a basic "play the game and say yep, seems about the same" smoke test, and I'm honestly not sure how I'd go about it offhand, but I do doubt that it matters. The old approach of writing everything in place to a caller-supplied pointer is from the 90s when CPUs performed in very different ways to what they do now, and returning smallish structs by value was a lot rarer.

Reply:
It would be better to get some hard numbers.  It's such an ubiquitous and essential part of FSO, that we should make sure it's not slowed down by this change.